esphome/components/hdc302x/hdc302x.cpp

```diff
@@ -0,0 +1,157 @@
+#include "hdc302x.h"
+
+#include "esphome/core/helpers.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace hdc302x {
+
+static const char *const TAG = "hdc302x.sensor";
+
+// Commands (per datasheet Table 7-4)
+static const uint8_t HDC302X_CMD_SOFT_RESET[2] = {0x30, 0xa2};
+static const uint8_t HDC302X_CMD_CLEAR_STATUS_REGISTER[2] = {0x30, 0x41};
+
+static const uint8_t HDC302X_CMD_TRIGGER_MSB = 0x24;
+static const uint8_t HDC302X_CMD_TRIGGER_LSB[] = {0x00, 0x0b, 0x16, 0xff};
+
+static const uint8_t HDC302X_CMD_HEATER_ENABLE[2] = {0x30, 0x6d};
+static const uint8_t HDC302X_CMD_HEATER_DISABLE[2] = {0x30, 0x66};
+
+void HDC302XComponent::setup() {
+  // Soft reset the device
+  ESP_LOGD(TAG, "Resetting I2C address: 0x%02X", this->address_);
+  if (this->write(HDC302X_CMD_SOFT_RESET, 2) != i2c::ERROR_OK) {
+    ESP_LOGE(TAG, "Resetting I2C device failed");
+    this->mark_failed("soft reset failed");
```

Comment:
`mark_failed` will already log the provided message
```suggestion
    this->mark_failed("soft reset failed");
```